### PR TITLE
refine loop run

### PR DESCRIPTION
**Pull request by skydoorkai**

### CODE REVIEW

File: elasticdl/python/master/master.py [hunk 1]
```diff
@@ -52,19 +52,30 @@ def run(self):
...
-                            InstanceManagerStatus.FINISHED
-                        )
-                    break
-                if self.pod_manager.all_workers_exited:
-                    raise Exception(
-                        "All workers exited but there also are",
-                        "unfinished tasks",
-                    )
-                time.sleep(30)
+            no_loop_run = not self.task_manager and not self.pod_manager
```
Comment (brightcoder01) [inline elasticdl/python/master/master.py]: I'm wondering whether the scenario exists that neither task_manager nor pod_manager is created.

Comment (skydoorkai) [reply elasticdl/python/master/master.py]: When users set arguments wrong.

Comment (brightcoder01) [reply elasticdl/python/master/master.py]: Maybe we can do some argument check and raise ValueError directly.

Comment (skydoorkai) [reply elasticdl/python/master/master.py]: Maybe we should assume that when it `run`, it should satisfy that at least one manager is running. 
Then we need to add an argument validation in `init` or `prepare`.

**Review by workingloong — DISMISSED** [review]

File: elasticdl/python/master/master.py [hunk 2]
```diff
@@ -128,3 +130,23 @@ def create_master_grpc_service(self, args):
             self.rendezvous_server,
             evaluation_service,
         )
+
+    def validate(self):
```
Comment (workingloong) [inline elasticdl/python/master/master.py:134]: Add an unit test for the method.

File: elasticdl/python/tests/master_test.py [hunk 3]
```diff
@@ -65,3 +65,34 @@ def test_master_run_and_stop(self):
...
+                master.validate()
+
+            self.arguments["need_elasticdl_job_service"] = "False"
+            args = self._get_args()
+            args = parse_master_args(args)
+            master = Master(args)
+            master.validate()
+
+            self.arguments["task_fault_tolerance"] = "False"
+            self.arguments["task_fault_tolerance"] = "True"
```
Comment (brightcoder01) [inline elasticdl/python/tests/master_test.py:91]: Why do wen set `task_fault_tolerance` twice?

Comment (skydoorkai) [reply elasticdl/python/tests/master_test.py:91]: Copy&Paste bug. Need to delete the 1st one.

**Review by workingloong — APPROVED** [review]

**Pull request merged**